### PR TITLE
feat: migrate projects to use store/context

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Original tickets: 

- WEB-671
- WEB-688

As part of improving our global state, we're gradually improving some pieces of shared state, right now we're focusing at the workspace projects and adding a collection of all projects which would be available for any subtree to access without re-fetching it.

## Test Plan

Go to the `Uncategorized` experiments list, select one or multiple experiments, select the action to `move` the experiments.

## Commentary (optional)

The modal will behave in the same way as in master, which has a known bug related to trying to select a different workspace from the dropdown, but, the immediate benefits from this implementation is the reduced amount of requests and the new global state.

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:157]: fix: this condition will never be true because on line 153 above you're only rendering this part if `workspaceId !== 1`. This condition should be changed back to `workspaceId === undefined ? ...`

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:58]: fix: because the initial workspaceId is 1 but the Uncategorized workspace is filtered out of the list, the initial state of the modal looks like this, a Workspace named "1" chosen that doesn't exist:
Screenshot 2022-11-30 at 15-00-12 Experiment 189 - Determined

**Review by ClaireNeveu — COMMENTED** [review]
LGTM. Please wait for @EmilyBonar's approval before merging

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:295]: nit: I think `projectId` does need to be in this dep array since you don't want the callback function to be calculated using an old value.

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
Looks pretty good, but some changes need to be made regarding moving experiments to the Uncategorized workspace

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx
Comment (ClaireNeveu) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:67]: This wasn't added in that PR, the line was just moved: https://github.com/determined-ai/determined/commit/95f704a00f1060d280baef236bf7678d7674f25c#diff-6bcd32a264fe358f263dea601fdc5a64beb55795a4e2689276245f843b653961L102

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx
Comment (ClaireNeveu) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:157]: workspaceId will also never be undefined so it seems like this is just dead code.

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx
Comment (EmilyBonar) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:67]: Ah, I see, it was actually added [here](https://github.com/determined-ai/determined/pull/5121/files). @mapmeld, was this intended?

Comment (mapmeld) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:67]: Maybe it should be preventing moves to archived workspaces?  In `api_experiment.go` it does prevent moving to/from archived workspaces and projects on the server.  I don't see anything about immutable there 😬 I think I botched that or was remembering an older conversation about how some people don't have access to Uncategorized in RBAC.

Comment (ClaireNeveu) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:67]: @mapmeld What should the correct behavior be then?

Comment (mapmeld) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:67]: I think replacing this and the `!w.immutable` in the projects listing logic with `!w.archived`, or adding `{archived: false}` to the useWorkspaces query, whichever makes sense

File: webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx
Comment (EmilyBonar) [reply webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx:58]: You're correct that the Uncategorized workspace has an id of 1. In your screenshot, the 1 in the URL is actually the  **project** id of 1, though that's a minor thing. The problem I was pointing out is that the text in the dropdown was "1", which is not the name of the Uncategorized workspace.

**Review by EmilyBonar — APPROVED** [review]
LGTM

**Pull request merged**
